services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -471,6 +471,25 @@ async def create_output_dirs(request_mode: CreateDirsRequestItem) -> None:
             dir_to_create.mkdir(parents=True, exist_ok=True)
 
 
+@containers_router.post(
+    "/containers/volumes:permissions",
+    summary="Makes permissions the same on all mounted volumes directories.",
+    response_class=Response,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def volumes_fix_permissions() -> None:
+    # NOTE: by creating a hidden file on all mounted volumes
+    # the same permissions are ensured and avoids
+    # issues when starting the services
+    mounted_volumes: MountedVolumes = get_mounted_volumes()
```

Comment:
No, the dynamic-sidecar needs first to start the container, once these are started and have this folders mounted you can finally create these files.

Reply:
A bit of a correction, I got confused (😕).

The director-v2 starts the `dynamic-sidecar`, which will mounts volumes to share data with the containers inside  `/dy-volumes` path. Before the dynamic-sidecar can continue to create the "containers" it has to create these files.

To answer your initial question: yes it could be done. It can be done after the boot. Not sure if it will work (in theory it should).

You might still want to know why was it done here. By design the dynamic-sidecar is build as an extension for the director-v2 which via this events picks the appropriate moment to run commands.

Reply:
As agreed after out mark. Will move the fix to the dynamic-sidecar startup.